pkg/roles/service_test.go

```diff
@@ -418,13 +427,37 @@ func TestServiceCreateRole(t *testing.T) {
 				},
 			)
 
+			if test.expected == nil {
+				mockOpenFGA.EXPECT().WriteTuples(gomock.Any(), gomock.Any()).Times(1).DoAndReturn(
+					func(ctx context.Context, tuples ...ofga.Tuple) error {
+						ps := make([]ofga.Tuple, 0)
+
+						ps = append(ps, *ofga.NewTuple(fmt.Sprintf("user:*"), CAN_VIEW_RELATION, fmt.Sprintf("role:%s", test.input.role)))
+
+						if !reflect.DeepEqual(ps, tuples) {
+							t.Errorf("expected tuples to be %v got %v", ps, tuples)
+						}
+
+						return test.secondExpected
+					},
+				)
+			}
+
 			if test.expected != nil {
 				mockLogger.EXPECT().Error(gomock.Any()).Times(1)
```

Comment:
try mintimes and maxtimes to remove the secondExpected

Reply:
ok seemsore complex than I thought, worth a chat

Reply:
Bundling all the writes in one go as you suggested will remove the need for a secondExpected